pep-9999.rst

```diff
@@ -81,9 +81,201 @@ includes stubs for Python's standard library and several third-party
 packages. These are usually distributed with type checkers and do not
 require separate installation.
 
+Style Guide
+===========
+
+Stub files should generally follow the Style Guide for Python Code (PEP 8)
+[#pep8]_. There are a few exceptions, outlined below, that take the
+different structure of stub files into account and are aimed to create
+more concise files.
+
+Maximum Line Length
+-------------------
+
+Type stubs should be limited to 130 characters per line.
+
+Blank Lines
+-----------
+
+Do not use empty lines between functions, methods, and fields, except to
+group them with one empty line. Use one empty line around classes, but do not
+use empty lines between body-less classes, except for grouping.
+
+Yes::
+
+    def time_func() -> None: ...
+    def date_func() -> None: ...
+
+    def ip_func() -> None: ...
+
+    class Foo:
+        x: int
+        y: int
+        def __init__(self) -> None: ...
+
+    class MyError(Exception): ...
+    class AnotherError(Exception): ...
+
+No::
+
+    def time_func() -> None: ...
+
+    def date_func() -> None: ...  # do no leave unnecessary empty lines
+
+    def ip_func() -> None: ...
+
+
+    class Foo:  # leave only one empty line above
+        x: int
+    class MyError(Exception): ...  # leave an empty line after the class
+
+Module Level Dunder Names
+-------------------------
+
+Do not include ``__all__`` in type stubs.
```

Comment:
I've been wondering about whether we should do this, actually. It seems helpful for type checkers to have `__all__` available, and I think mypy actually uses it.

Reply:
This was copied from https://github.com/python/mypy/wiki/Creating-Stubs-For-Python-Modules, originally written by @JukkaL. There is a bit more rationale for it there. We actually have a second point in the Wiki under the "supported stuff" heading:

> Is `__all__` meaningful in a stub? What does it do?

I have no opinion about this.

Reply:
I removed `__all__` from this section and opened #20 to discuss this for the "what belongs in a stub" section.